test/test_dynamic_shapes.py

```diff
@@ -0,0 +1,292 @@
+import torch
+from torch.utils._pytree import tree_map
+from contextlib import contextmanager
+aten = torch.ops.aten
+import unittest
+
+from torch.testing._internal.common_utils import run_tests, TestCase
+from torch.testing._internal.jit_utils import JitTestCase
+
+try:
+    import sympy
+    HAS_SYMPY = True
+except ImportError:
+    HAS_SYMPY = False
+skipIfNoSympy = unittest.skipIf(not HAS_SYMPY, "no sympy")
+
+from torch._C import _disabled_torch_function_impl
+
+@contextmanager
+def no_dispatch():
+    guard = torch._C._DisableTorchDispatch()  # type: ignore[attr-defined]
+    try:
+        yield
+    finally:
+        del guard
```

Comment:
This finally is in a reasonable module no need to duplicate it

Reply:
I actually didn't need with the latest @Chillee 's PoC.